plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -417,6 +418,21 @@ instance HasConstantIn DefaultUni term => ReadKnownIn DefaultUni term Word8 wher
                _       -> throwing_ _EvaluationFailure
     {-# INLINE readKnown #-}
 
+-- | This allows us to transparently use values of type LiteralByteSize as
+-- built-in Integers but with a different size measure.
+instance KnownTypeAst tyname DefaultUni LiteralByteSize where
+    toTypeAst _ = toTypeAst $ Proxy @Integer
+
+-- See Note [Integral types as Integer].
+instance HasConstantIn DefaultUni term => MakeKnownIn DefaultUni term LiteralByteSize where
```

Comment:
You could probably simply use newtype-deriving for all the three instances?

Reply:
I think I considered that and then decided that it would be easier just to mimic what it does for the other integral types.